Sourcery/Generating/Template/Swift/SwiftTemplate.swift

```diff
@@ -0,0 +1,192 @@
+            }
+
+            let code = component.substring(to: endIndex.lowerBound)
+            if code.hasPrefix("=") {
+                let codeStartIndex = code.index(code.startIndex, offsetBy: 1)
+                let realCode = code.substring(from: codeStartIndex)
+                sourceFile.append("\n print(\"\\(" + realCode + ")\", terminator: \"\");")
+            } else {
+                sourceFile.append(code)
+            }
+
+            let encodedPart = component.substring(from: endIndex.upperBound)
+            sourceFile.append(("\n print(\"") + encodedPart.stringEncoded + "\", terminator: \"\");")
+            for _ in 0 ..< encodedPart.numberOfLineSeparators {
+                sourceFile.append("\n")
+            }
+        }
+
+        return sourceFile.joined(separator: "")
+    }
+
+    func render(types: [Type], arguments: [String: NSObject]) throws -> String {
+        let context = GenerationContext(types: types, arguments: arguments)
+        let swiftCode = try SwiftTemplate.generateSwiftCode(templateContent: try sourcePath.read(), path: sourcePath)
+
+        let compilationDir = Path.cleanTemporaryDir(name: "build")
+
+        let runtimeFiles = try SwiftTemplate.swiftTemplatesRuntime.children().map { file in
+            return file.description
+        }
+
+        let mainFile = compilationDir + Path("main.swift")
+        let binaryFile = compilationDir + Path("bin")
+
+        let runableCode = "extension GenerationContext { override func generate() { " + swiftCode + " } }; run();"
+
+        try mainFile.write(runableCode)
+
+        let serializedContextPath = compilationDir + Path("context.bin")
+
+        let serializedContext = NSKeyedArchiver.archivedData(withRootObject: context)
+        try serializedContextPath.write(serializedContext)
+
+        #if DEBUG
+            // this is a sanity check, deserialized object should be equal to initial object
+            let diff = context.diffAgainst(NSKeyedUnarchiver.unarchiveObject(with: serializedContext))
+            if !diff.isEmpty {
+                print(diff.description)
+            }
+            assert(diff.isEmpty)
+        #endif
+
+        let arguments = [mainFile.description] + runtimeFiles + ["-Onone", "-module-name", "Sourcery"] + ["-o"] + [binaryFile.description]
+
+        let compilationResult = try Process.runCommand(path: "/usr/bin/swiftc",
+                                            arguments: arguments,
+                                            environment: [:])
+
+        if !compilationResult.error.isEmpty {
+            #if DEBUG
+                let command = "/usr/bin/swiftc " + arguments.map { "\"\($0)\"" }.joined(separator: " ")
+                print(command)
+            #endif
+            throw SwiftTemplateParsingError.compilationError(path: binaryFile, error: compilationResult.error)
+        }
+
+        let result = try Process.runCommand(path: binaryFile.description,
```

Comment:
So if I get it correctly we encode context with all types information, then we pass the path to this archive to another swift file in which we insert code that has `<%= %>` replaced with prints and that decodes the archive and runs the printing code we inserted in it. And then we get the output of all these prints as a result. Right?
Do we somehow handle the issue with empty lines when removing control code (https://github.com/kylef/Stencil/pull/32) or we don't have it at all?